src/core_atmosphere/dynamics/mpas_atm_boundaries.F

```diff
@@ -225,15 +289,58 @@ subroutine mpas_atm_update_bdy_tend(clock, streamManager, block, firstCall, ierr
 
 
             dt = 1.0_RKIND / dt
-            lbc_tend_u(:,:) = (u(:,:) - lbc_tend_u(:,:)) * dt
-            lbc_tend_ru(:,:) = (ru(:,:) - lbc_tend_ru(:,:)) * dt
-            lbc_tend_rho_edge(:,:) = (rho_edge(:,:) - lbc_tend_rho_edge(:,:)) * dt
-            lbc_tend_w(:,:) = (w(:,:) - lbc_tend_w(:,:)) * dt
-            lbc_tend_theta(:,:) = (theta(:,:) - lbc_tend_theta(:,:)) * dt
-            lbc_tend_rtheta_m(:,:) = (rtheta_m(:,:) - lbc_tend_rtheta_m(:,:)) * dt
-            lbc_tend_rho_zz(:,:) = (rho_zz(:,:) - lbc_tend_rho_zz(:,:)) * dt
-            lbc_tend_rho(:,:) = (rho(:,:) - lbc_tend_rho(:,:)) * dt
-            lbc_tend_scalars(:,:,:) = (scalars(:,:,:) - lbc_tend_scalars(:,:,:)) * dt
+
+            !$acc parallel default(present)
+            !$acc loop gang vector collapse(2)
+            do iEdge=1,nEdges+1
+                do k=1,nVertLevels
+                    lbc_tend_u(k,iEdge) = (u(k,iEdge) - lbc_tend_u(k,iEdge)) * dt
+                    lbc_tend_ru(k,iEdge) = (ru(k,iEdge) - lbc_tend_ru(k,iEdge)) * dt
+                end do
+            end do
+
+            !$acc loop gang vector collapse(2)
+             do iEdge=1,nEdges+1
+                 do k=1,nVertLevels
+                     lbc_tend_rho_edge(k,iEdge) = (rho_edge(k,iEdge) - lbc_tend_rho_edge(k,iEdge)) * dt
```

Comment:
Was there a specific reason for computing `lbc_tend_u` and `lbc_tend_ru` in the same loop, but using a separate loop for `lbc_tend_rho_edge`?

Reply:
I wasn't sure if splitting up the loops would result in better performance on GPUs, but I'll just keep it simple and consolidate all similar loops for now and investigate optimizations later.